ttnn/cpp/ttnn/operations/eltwise/binary_backward/device/binary_backward_op.hpp

```diff
@@ -41,6 +41,15 @@ enum class BinaryBackwardOpType {
     LERP_BW,
     MUL_BW,
 };
+struct BinaryBackwardFunction{
+static std::function<std::vector<ttnn::Tensor>(const Tensor&, const Tensor&, const Tensor&, const MemoryConfig&)> get_function_type1(BinaryBackwardOpType OpType);
+static std::function<std::vector<ttnn::Tensor>(const Tensor&, const Tensor&, const Tensor&, float, const MemoryConfig&)> get_function_type1_w_float(BinaryBackwardOpType OpType);
```

Comment:
overall I'd create a unordered_map<OptType, variant<all function types>> and simplify all this functions.